doc/architecture/decisions/0006-remove-govukpayment-from-session-data.md

```diff
@@ -0,0 +1,40 @@
+# 6. Remove `GovUkPayment` from session data
+
+Date: 2023-07-03
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, we store the `GovUkPayment` object alongside the session data in the `lowcal_storage.data` column. This is largely done for convenience - it allows us to easily find out the payment status of a session. With recent changes to the structure of breadcrumbs, it makes sense to re-evaluate how this data is stored and referred to.
+
+## Decision
+
+We should not store the `GovUkPayment` object, but look up the status of a session in one of the two following ways when required - 
+ - Directly via the GovPay API
+ - Via our `payment_status` audit table which can be joined to a session
+
+This object can then be stored temporarily in the application's memory during the user's active session (Zustand store).
+
+## Consequences
+
+### Required code changes
+We will need to make a series of changes to how we currently interact with this object, listed below - 
+
+| Interaction | Context | Required change | Consequence |
+|--|--|--|--|
+| Pay component | `GovUkPayment` is required here to get status and `next_url` links | On component load, fetch GovPay ID via audit table and query GovPay API. This could be wrapped in a `planx-core` function `session.get`GovUkPayment`(sessionId)`. <br/><br/> We would need to handle Hasura permissions here to ensure a public user can access `payment_status.gov_pay_id` for their session ([Hasura docs](https://hasura.io/docs/latest/auth/authorization/permissions/row-level-permissions/#relationships-in-permissions)) | Increased load time on pay component for additional queries |
```

Comment:
Question: in this in-flow Pay scenario, are we still mutating session data upon successful payment? Not to store the `GovUkPayment` object, but to update the session `breadcrumbs` to reflect the Pay node? 

Is it worth noting any specific mutations for this scenario as well as queries?

Reply:
I'll add this in to be explicit 👍 

In this scenario the Pay component would still generate a breadcrumb exactly as it currently does, it would simply not be adding the additional `session.data.govUkPayment` to the passport.